sycl/include/sycl/accessor.hpp

```diff
@@ -1141,34 +1141,34 @@ class __SYCL_EBO __SYCL_SPECIAL_CLASS __SYCL_TYPE(accessor) accessor :
   accessor(const detail::AccessorImplPtr &Impl)
       : detail::AccessorBaseHost{Impl} {}
 
-  id<3> &getOffset() {
+  id<3> getOffset() {
     if constexpr (IsHostBuf)
-      return MAccData->MOffset;
+      return MAccData ? MAccData->MOffset : id<3>();
     else
       return AccessorBaseHost::getOffset();
   }
 
   range<3> &getAccessRange() { return AccessorBaseHost::getAccessRange(); }
-  range<3> &getMemoryRange() {
+  range<3> getMemoryRange() {
     if constexpr (IsHostBuf)
-      return MAccData->MMemoryRange;
+      return MAccData ? MAccData->MMemoryRange : range(0, 0, 0);
     else
       return AccessorBaseHost::getMemoryRange();
   }
   void *getPtr() { return AccessorBaseHost::getPtr(); }
 
-  const id<3> &getOffset() const {
+  const id<3> getOffset() const {
     if constexpr (IsHostBuf)
-      return MAccData->MOffset;
+      return MAccData ? MAccData->MOffset : id<3>();
     else
       return AccessorBaseHost::getOffset();
   }
   const range<3> &getAccessRange() const {
     return AccessorBaseHost::getAccessRange();
   }
-  const range<3> &getMemoryRange() const {
+  const range<3> getMemoryRange() const {
```

Comment:
Now that these no longer return references, is there a reason to have a non-const version?

Reply:
I guess no [70e5b6f](https://github.com/intel/llvm/pull/8557/commits/70e5b6f6b43a28864d48fcf71846d5a7f7bc0c4f)